### PR TITLE
Cross compile for 2.11.7, RegExValidator and InCaseValidator

### PR DESCRIPTION
**Pull request by EdgeCaseBerg**

### CODE REVIEW

File: src/main/scala/com/plameno/validation/CommonValidators.scala [hunk 1]
```diff
@@ -23,12 +25,16 @@ case class MaxLengthValidator(s: String, maxLen: Int, customError: Option[String
   def isValid = s.length <= maxLen
 }
 
-// TODO: RegexValidator with compiled pattern. And reuse it for email validation.
+case class RegexValidator(s: String, re: Regex, customError: Option[String] = None)
+  extends Validator(customError) {
+  override val defaultError = s"Regular expression ${re} could not be matched"
```
Comment (siman) [inline src/main/scala/com/plameno/validation/CommonValidators.scala]: Surround ${re} with quotes '' to have more user friendly message:
`s"Regular expression '${re}' could not`...

File: src/test/scala/com/plameno/validation/ValidationSpec.scala [hunk 2]
```diff
@@ -32,6 +32,17 @@ class ValidationSpec extends FlatSpec {
...
   }
+
+  it should "validate with regular expressions" in {
+    assert(RegexValidator("hello world!", "^hello".r).isValid)
+    val errorMsg = "No x marks the spot!"
+    assert(RegexValidator("beep", "x".r, Some(errorMsg)).validate == Some(errorMsg))
+  }
+
+  it should "validate any supplied function in the InCaseValidator" in {
+    assert(InCaseValidator(2, (x : Int) => {x % 2 == 0}).isValid)
```
Comment (siman) [inline src/test/scala/com/plameno/validation/ValidationSpec.scala:43]: Here should be 2 test cases:
1. valid (you already made it)
2. invalid (TODO)

File: project/Build.scala [hunk 3]
```diff
@@ -12,6 +12,7 @@ object BuildSettings {
     organization := "com.plameno",
     version := "0.1-SNAPSHOT",
     scalaVersion := "2.10.2",
+    crossScalaVersions := Seq("2.11.7"),
```
Comment (siman) [inline project/Build.scala:15]: :cool:

File: src/main/scala/com/plameno/validation/CommonValidators.scala (hunk 1 above)
Comment (EdgeCaseBerg) [inline src/main/scala/com/plameno/validation/CommonValidators.scala]: Good call. Addressed in a1f3e05

File: src/test/scala/com/plameno/validation/ValidationSpec.scala (hunk 2 above)
Comment (EdgeCaseBerg) [inline src/test/scala/com/plameno/validation/ValidationSpec.scala:43]: Good point.
addressed in cb40ca7

File: project/Build.scala (hunk 3 above)
Comment (EdgeCaseBerg) [inline project/Build.scala:15]: Isn't it? :) 
You can test both versions by adding the `+` sign before an SBT command.

**Comment by siman** [comment]
Good job! Just wondering where do you use these validations?

**Comment by EdgeCaseBerg** [comment]
I was looking for a simple scala validator library last night and stumbled across yours. I had dealt with a [different validation library](https://github.com/wix/accord) before that was a bit more complicated  and promised custom error messages, but I couldn't get the error messaging to work. 

Your code is way easier to understand than the accord stuff and I noticed the To-do's and thought I might contribute a little :)

**Pull request merged**